azurerm/resource_arm_batch_pool.go

```diff
@@ -0,0 +1,476 @@
+		}
+
+		log.Printf("[INFO] stopping the pending resize operation on this pool...")
+		if _, err = client.StopResize(ctx, resourceGroup, accountName, poolName); err != nil {
+			return fmt.Errorf("Error stopping resize operation for Batch pool %q (Resource Group %q): %+v", poolName, resourceGroup, err)
+		}
+
+		// waiting for the pool to be in steady state
+		log.Printf("[INFO] waiting for the pending resize operation on this pool to be stopped...")
+		isSteady := false
+		for !isSteady {
+			resp, err = client.Get(ctx, resourceGroup, accountName, poolName)
+			if err != nil {
+				return fmt.Errorf("Error retrieving the Batch pool %q (Resource Group %q): %+v", poolName, resourceGroup, err)
+			}
+
+			isSteady = resp.AllocationState == batch.Steady
+			time.Sleep(time.Minute * 2)
+			log.Printf("[INFO] waiting for the pending resize operation on this pool to be stopped... New try in 2 minutes...")
+		}
+	}
+
+	parameters := batch.Pool{
+		PoolProperties: &batch.PoolProperties{},
+	}
+
+	scaleMode := azure.BatchPoolScaleMode(d.Get("scale_mode").(string))
+	if scaleMode == azure.BatchPoolAutoScale {
+		autoScaleEvaluationInterval := d.Get("autoscale_evaluation_interval").(string)
+		autoScaleFormula := d.Get("autoscale_formula").(string)
+
+		if autoScaleFormula == "" {
+			return fmt.Errorf("Error: when scale mode is set to Auto, auto_scale formula cannot be empty")
+		}
+
+		parameters.PoolProperties.ScaleSettings = &batch.ScaleSettings{
+			AutoScale: &batch.AutoScaleSettings{
+				EvaluationInterval: &autoScaleEvaluationInterval,
+				Formula:            &autoScaleFormula,
+			},
+		}
+	} else if scaleMode == azure.BatchPoolFixedScale {
+		targetDedicatedNodes := int32(d.Get("target_dedicated_nodes").(int))
+		targetLowPriorityNodes := int32(d.Get("target_low_priority_nodes").(int))
+		resizeTimeout := d.Get("resize_timeout").(string)
+
+		parameters.PoolProperties.ScaleSettings = &batch.ScaleSettings{
+			FixedScale: &batch.FixedScaleSettings{
+				ResizeTimeout:          &resizeTimeout,
+				TargetDedicatedNodes:   &targetDedicatedNodes,
+				TargetLowPriorityNodes: &targetLowPriorityNodes,
+			},
+		}
+	} else {
+		return fmt.Errorf("Error: scale mode should be either AutoScale of FixedScale")
+	}
+
+	if _, err = client.Update(ctx, resourceGroup, accountName, poolName, parameters, ""); err != nil {
+		return fmt.Errorf("Error creating Batch pool %q (Resource Group %q): %+v", poolName, resourceGroup, err)
+	}
+
+	read, err := client.Get(ctx, resourceGroup, accountName, poolName)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Batch pool %q (Resource Group %q): %+v", poolName, resourceGroup, err)
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read Batch pool %q (resource group %q) ID", poolName, resourceGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmBatchPoolRead(d, meta)
+}
+
+func resourceArmBatchPoolDelete(d *schema.ResourceData, meta interface{}) error {
```

Comment:
Could we put this under the read function?

Reply:
Done.